Publish temperatures for monitoring overheating

christian-rauch opened the pull request:
Publish all temperature readings as `TemperatureArray` on topic `temperature`. The names of the temperature sensors are currently hard coded in `temperature_names_` since there is no way to retrieve them from the API.

This PR depends on PR https://github.com/ipa320/schunk_modular_robotics/pull/192 and already includes its changes. I am happy to rebase this PR on `indigo_dev` if it should be merged first.

christian-rauch committed:
enable C++11

christian-rauch committed:
publish temperatures

wxmerkt commented:
@christian-rauch There is already a Temperature message in ``sensor_msgs``: http://docs.ros.org/api/sensor_msgs/html/msg/Temperature.html

christian-rauch commented:
@wxmerkt I saw this, but it does not include an identifier. By using `sensor_msgs/Temperature` we would need to publish each temperature reading on a separate topic. I could not find a commonly used message for a list of temperatures with identifiers similar to `sensor_msgs/JointState`.

fmessmer commented on schunk_sdh/msg/TemperatureArray.msg:
```diff
@@ -0,0 +1,2 @@
+Header header
+Temperature[] temperature_list
```
how about using http://docs.ros.org/api/sensor_msgs/html/msg/JointState.html as an inspiration?
i.e. header + string vector name + double vector temperature?

christian-rauch replied:
I found having a list like `{ {name1, value1}, {name2, value2}, ... }` (e.g. a list of key-value pairs like `std::map`) makes iterating the list and searching for specific `name`s easier. In a list like `{ {name1, name2, ...}, {value1, value2, ...} }`, you would first need to find the index corresponding to a `name` to access its `value` from the other list.

But if you prefer a structure like `JointState`, I will change the message type.

fmessmer replied:
I don't have a preference, as we are not using the hardware atm...
I was just throwing in an idea...I leave it to you to decide...maybe @wxmerkt has an opinion here, too?

wxmerkt replied:
I am happy with either way; while being similar to ``sensor_msgs/JointState`` is the more standard approach for message types, I like the motivation with iterators though.

christian-rauch commented:
I changed the message definition to be similar to `sensor_msgs/JointState` which probably increases the likelihood of becoming a standard ROS message later. I also rebased in `indigo_dev` because it is a less intrusive change.
